Eevee.Sleep.Bot/Handlers/EventHandlers/GuildMemberUpdatedEventHandler.cs

```diff
@@ -86,6 +86,15 @@ IUser user
         );
     }
 
+    private static async Task HandleRolesAdded(ulong userId, ulong[] addedRoles) {
```

Comment:
Should be able to change this to a simple call of `DiscordRoleContoller.AddRoles(userId, addedRoles);` then let the DB handle the de-duplication (that's why `$pushToSet`).

Just need to make sure that the `addedRoles` above are tracked ones. Untracked ones shouldn't get passed in the call of `AddRoles()`. This can be done by set intersection of finding the tracked roles and the unfiltered added roles.

Reply:
What is passed here are new roles that should be added (e.g., level-up roles).
This is always untracked role.
How do we determine that this is a role that should be added?
Or should we not automatically add this?

Reply:
> What is passed here are new roles that should be added (e.g., level-up roles).
If these roles are not in our list of tracking roles, then they should never gets added, if the addition you mean is the role ownership of an user.

> How do we determine that this is a role that should be added?
If you mean the addition of the ownership, this is why I said we need another mongo collection to record what roles are we tracking, so we can know if the roles here from `addedRoles` are tracked and know if we should record the ownership of the role.

Reply:
Obviously, we needed a new schema...
I'll fix it.

Reply:
Changed to be managed in a separate table
[758997e](https://github.com/RaenonX-PokemonSleep/eevee-sleep-discord-bot/pull/25/commits/758997e297da2592193f3b6db9effa5b52a79d60)